extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSTableSuite.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.spark.connector.tpcds
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+
+import org.apache.kyuubi.KyuubiFunSuite
+import org.apache.kyuubi.spark.connector.tpcds.LocalSparkSession.withSparkSession
+
+class TPCDSTableSuite extends KyuubiFunSuite {
+
+  test("useAnsiStringType (true,false)") {
+    Seq(true, false).foreach(key => {
+      val sparkConf = new SparkConf().setMaster("local[*]")
+        .set("spark.ui.enabled", "false")
+        .set("spark.sql.catalogImplementation", "in-memory")
+        .set("spark.sql.catalog.tpcds", classOf[TPCDSCatalog].getName)
+        .set("spark.sql.catalog.tpcds.useAnsiStringType", key.toString)
+      withSparkSession(SparkSession.builder.config(sparkConf).getOrCreate()) { spark =>
+        val rows = spark.sql("desc tpcds.sf1.call_center").collect()
```

Comment:
How about using `spark.table("tpcds.sf1.call_center").schema`?

Reply:
Because Spark3.1([SPARK-33480](https://issues.apache.org/jira/browse/SPARK-33480)) and above will replace char/varchar with string and put the type information in field metadata.

org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation#create
```scala
    // The v2 source may return schema containing char/varchar type. We replace char/varchar
    // with "annotated" string type here as the query engine doesn't support char/varchar yet.
    val schema = CharVarcharUtils.replaceCharVarcharWithStringInSchema(table.schema)
    DataSourceV2Relation(table, schema.toAttributes, catalog, identifier, options)
```

![image](https://user-images.githubusercontent.com/3898450/167544529-3bae345b-1fdb-423a-a6d2-e6d23b444786.png)